### DISCUSSION TITLE
Kernel crashes with 001_getting_started

### DISCUSSION THREAD

**Discussion by lweggartner** [open-edge-platform/anomalib#2001] [Q&A] [answered]
Hey all,

I am at the very beginning of my anomaly project. I just installed the anomalib package and am trying to run the "001_getting_started" notebook. Here I am already facing some difficulties.

My kernel seems to always crash within the training cell. I am not really sure why. I am running it on a lenovo T470 on CPU and have already reduced the batch_size to one. Could it possibly be due to some troubles regarding the lightningModule?

Has anyone faced similar troubles or any idea on how to solve this?

Code:
```
# start training
engine = Engine(task=TaskType.SEGMENTATION)
engine.fit(model=model, datamodule=datamodule)
```
Output:

> GPU available: False, used: False
TPU available: False, using: 0 TPU cores
IPU available: False, using: 0 IPUs
HPU available: False, using: 0 HPUs
`Trainer(val_check_interval=1.0)` was configured so validation will run at the end of the training epoch..
F1Score class exists for backwards compatibility. It will be removed in v1.1. Please use BinaryF1Score from torchmetrics instead
F1Score class exists for backwards compatibility. It will be removed in v1.1. Please use BinaryF1Score from torchmetrics instead
/home/weu4rt/anaconda3/envs/keras/lib/python3.11/site-packages/lightning/pytorch/core/optimizer.py:180: `LightningModule.configure_optimizers` returned `None`, this fit will run with no optimizer

>   | Name                  | Type                     | Params
> -------------------------------------------------------------------
> 0 | model                 | PadimModel               | 2.8 M 
> 1 | _transform            | Compose                  | 0     
> 2 | normalization_metrics | MinMax                   | 0     
> 3 | image_threshold       | F1AdaptiveThreshold      | 0     
> 4 | pixel_threshold       | F1AdaptiveThreshold      | 0     
> 5 | image_metrics         | AnomalibMetricCollection | 0     
> 6 | pixel_metrics         | AnomalibMetricCollection | 0     
> -------------------------------------------------------------------
> 2.8 M     Trainable params
> 0         Non-trainable params
> 2.8 M     Total params
> 11.131    Total estimated model params size (MB)
> /home/weu4rt/anaconda3/envs/keras/lib/python3.11/site-packages/lightning/pytorch/trainer/connectors/data_connector.py:441: 
> The 'train_dataloader' does not have many workers which may be a bottleneck. Consider increasing the value of the num_workers argument to num_workers=3 in the DataLoader to improve performance.
> /home/weu4rt/anaconda3/envs/keras/lib/python3.11/site-packages/lightning/pytorch/trainer/connectors/data_connector.py:441: 
> The 'val_dataloader' does not have many workers which may be a bottleneck. Consider increasing the value of the num_workers argument to num_workers=3 in the DataLoader to improve performance.

> Training:` |          | 0/? [00:00<?, ?it/s]

> /home/weu4rt/anaconda3/envs/keras/lib/python3.11/site-packages/lightning/pytorch/loops/optimization/automatic.py:129: `training_step` returned `None`. If this was on purpose, ignore this warning...

> **The Kernel crashed while executing code in the current cell or a previous cell. 
Please review the code in the cell(s) to identify a possible cause of the failure. 
Click [here](https://aka.ms/vscodeJupyterKernelCrash) for more info. 
View Jupyter [log](command:jupyter.viewOutput) for further details.**

**Comment by abc-125** [comment, marked as answer]
Hi, can you try this code to see if the problem might be the memory requirements? And it is better to make `batch_size` at least 4.
```
# Get the model and datamodule
model = Fastflow()
datamodule = MVTec(num_workers=0)
```

Also, can you please show the Jupyter log?

**Reply by lweggartner** [reply]
Thank you for help. Fastflow works fine :)
